### PR TITLE
add graph embeddings module

### PR DESCRIPTION
**Pull request by Bin-Go2**
Users can use `kgtk graph-embeddings OPTIONS` to get the embedding result of graph tsv

### CODE REVIEW

File: kgtk/cli/graph_embeddings.py [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+        subprocess_init = SubprocessInitializer()
+        input_edge_paths = [tmp_tsv_path] 
+
+        convert_input_data(
+            config.entities,
+            config.relations,
+            config.entity_path,
+            config.edge_paths,
+            input_edge_paths,
+            TSVEdgelistReader(lhs_col=0, rel_col=1, rhs_col=2),
```
Comment (filievski) [inline kgtk/cli/graph_embeddings.py:408]: this line assumes that the columns are in position 0, 1, 2 but this might not be the case

**Review by filievski — COMMENTED** [review]
The following should be adapted:
* the documentation should not include a 'prerequisite' section. Users should install all prerequisites during the installation of the overall toolkit.
* For each of the examples below, we should show a brief output.
* (minor) in example 1, use -i or --input-file instead of '<'

**Comment by filievski** [comment]
This command is not working yet. 

```
INFO:torchbiggraph:Loading entity counts...
INFO:torchbiggraph:Creating workers...
INFO:torchbiggraph:Initializing global model...
INFO:torchbiggraph:Starting epoch 1 / 100, edge path 1 / 1, edge chunk 1 / 1
INFO:torchbiggraph:Edge path: tmp/output/edges_partitioned
INFO:torchbiggraph:still in queue: 0
INFO:torchbiggraph:Swapping partitioned embeddings None ( 0 , 0 )
INFO:torchbiggraph:( 0 , 0 ): Loading entities
OMP: Error #15: Initializing libiomp5.dylib, but found libomp.dylib already initialized.
OMP: Hint This means that multiple copies of the OpenMP runtime have been linked into the program. That is dangerous, since it can degrade performance or cause incorrect results. The best thing to do is to ensure that only a single OpenMP runtime is linked into the process, e.g. by avoiding static linking of the OpenMP runtime in any library. As an unsafe, unsupported, undocumented workaround you can set the environment variable KMP_DUPLICATE_LIB_OK=TRUE to allow the program to continue to execute, but that may cause crashes or silently produce incorrect results. For more information, please see http://www.intel.com/software/products/support/.
/Users/filipilievski/mcs/kgtk/kgtk/exceptions.py:66: UserWarning: Please raise KGTKException instead of <class 'NameError'>
  warnings.warn('Please raise KGTKException instead of {}'.format(type_))
KGTKException found
```

**Comment by Bin-Go2** [comment]
Hi，professor, this issue appears when at 10-22 report, the solution is
1.Initializing libiomp5.dylib, but found libiomp5.dylib already initialized.
solution:
import os
os.environ[&quot;KMP_DUPLICATE_LIB_OK&quot;]=&quot;TRUE&quot;

I will update the codes and then pull a new request.

Before I pull a new request, let me just clarify what I need to do:
1. delete prerequisite part
2. add examples for input and output
3.  (minor) in example 1, use -i or --input-file instead of '<'
4. solve the issues

Do I have any other things to update?

File: kgtk/cli/graph_embeddings.py (hunk 1 above)
Comment (Bin-Go2) [reply kgtk/cli/graph_embeddings.py:408]: Yeah, that's fine, cause we use the KGTKReader to read the nodes and relation, and we do an intermediate process to make the node1 at position 0, relation at position 1, and node2 at position 2

**Comment by filievski** [comment]
Indeed, this seems sufficient for the next pull request. Especially, making sure that the command runs in the dev branch. 
Once I can run the command and we start using it, there might be other things that come up.

**Comment by Bin-Go2** [comment]
OK, Got it.

**Pull request merged**